### ISSUE TITLE
Second order derivative in neural network w.r.t. input encounter all zero

### ISSUE THREAD

**Issue by uiyo** [pytorch/pytorch#4896]
`import torch`
`from torch.autograd import grad,Variable`

`input = Variable(torch.rand(1,3,224,224), requires_grad=True).cuda()`
`w = Variable(torch.rand(1,802816)).cuda()`
`net = vgg16_withoutfc().cuda()`
`output = net(input)`
`jacob = grad(output, input, grad_outputs=w, create_graph=True)[0]`
`y = grad(jacob.sum(), input, create_graph=True)[0]`

the values of y somehow are all zero
while in none-neural network case

`x = Variable(torch.rand(2,2), requires_grad=True).cuda()`
`z = Variable(torch.rand(2,2)).cuda()`
`n = x**2`
`jacob = grad(n, x, grad_outputs=z, create_graph=True)[0]`
`y = grad(jacob.sum(), x, create_graph=True)[0]`

the values of y are all normal
How to fix this?

**Comment by apaszke** [comment]
There is no "fix", zero is the correct answer in this case. Gradient of the loss with respect to the input doesn't depend on the value of that input, so it is zero.

**Comment by uiyo** [comment]
then how can i compute the second order gradients of network respect to the inputs? i could not find an example anywhere, please, i'm in emergency, this has taken me too long.

**Comment by uiyo** [comment]
Another thing is weird 
if i use resnet18 instead of vgg16 it's ok i  get the right gradient
:) by the way, we found tf vgg16 cannot either

**Comment by uiyo** [comment]
We found that actually vgg16 is a linear model, so its second order derivatives are all zero. if we modified the network to resnet18 this will work. somehow stupid question, anyway thank you for the time.
